docs/basic/getting-started/forward-create-ref.md

```diff
@@ -28,6 +28,20 @@ export const FancyButton = React.forwardRef<Ref, Props>((props, ref) => (
 ));
 ```
 
+`forwardRef` alternative without mutability:
```

Comment:
i dont understand what "without mutability" means. do you mind elaborating in here? how does this materially differ from the example above?

Reply:
`React.forwardRef<HTMLElement, Props>` assigns the type `((instance: T | null) => void) | MutableRefObject<T | null> | null` to the `ref` parameter. It's a `MutableRefObject<T | null>`, unlike with `React.Ref<T>`. The latter returns `((instance: T | null) => void) | RefObject<T> | null`, which seems to be an improvement.

Reply:
> which seems to be an improvement.

We should explain this in the cheatsheet where this would be an improvement. It would always be an improvement then you should propose the change to `@types/react`.

Reply:
alright @kripod, i understand where you are coming from, and agree with @eps1lon that we should fix this in `@types`. wanna open up an issue over there?

Reply:
Thank you for the discussion!

It appears that [an opposing issue](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/39062) was raised and [a PR has been merged](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43265) to make the `ref` parameter of `forwardRef` mutable.

Reply:
wow. we both learned something today...